File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())  return false;
        unordered_map<char,int> mpp1;
        for(char ch: s) mpp1[ch]++;
        for(char ch: t){
            if(!mpp1.count(ch)) return false; 
            mpp1[ch]--;
            if(mpp1[ch]==0) mpp1.erase(ch);
        }
        return mpp1.empty();
    }
};

File: 242-valid-anagram/valid-anagram.py
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mpp=Counter(s)
        for ch in t:
            if not ch in mpp:
                return False
            mpp[ch]-=1
            if mpp[ch]==0:
                del mpp[ch]
        
        return not mpp